src/codegen/codegen_c_visitor.cpp

```diff
@@ -1248,8 +1248,12 @@ void CodegenCVisitor::print_memory_allocation_routine() const {
     printer->add_newline(2);
     auto args = "size_t num, size_t size, size_t alignment = 16";
     printer->fmt_start_block("static inline void* mem_alloc({})", args);
-    printer->add_line("void* ptr;");
-    printer->add_line("posix_memalign(&ptr, alignment, num*size);");
+    printer->add_line("size_t fill = 0;");
+    printer->start_block("if (num % alignment != 0)");
+    printer->add_line("size_t multiple = num / alignment;");
+    printer->add_line("fill = alignment * (multiple + 1) - num;");
```

Comment:
I think the formula following formula for `ceil(num / alignment)` could come in handy:
```
(num*size + alignment - 1) / alignment) * alignment
```
(allows you to remove the `if` condition).